unit-tests/src/test/java/org/eclipse/collections/impl/bimap/mutable/HashBiMapTest.java

```diff
@@ -102,4 +105,22 @@ public void withKeysValues()
         Assert.assertSame(map, map44);
         Assert.assertSame(map3, map4);
     }
+
+    @Test
+    public void forcePut_inverseKeyPreservation()
+    {
+        Key key1 = new Key("1");
+        Key key2 = new Key("2");
+
+        Key value1 = new Key("abc");
+
+        Key duplicateOfKey1 = new Key("1");
+
+        HashBiMap<Key, Key> biMap = this.newMapWithKeysValues(key1, value1, key2, new Key("xyz"));
+
+        biMap.forcePut(duplicateOfKey1, new Key("xyz"));
+
+        Assert.assertSame(key1, Iterate.getFirst(biMap.entrySet()).getKey());
+        Assert.assertSame(key1, Iterate.getFirst(biMap.inverse().entrySet()).getValue());
+    }
```

Comment:
I realized that we need to assertSame on the value as well. Sorry for the confusion. I think this should be good:

`@Test
    public void forcePut_inverseKeyAndValuePreservation()
    {
        Key key1 = new Key("1");
        Key value2 = new Key("xyz");

        HashBiMap<Key, Key> biMap = this.newMapWithKeysValues(key1, new Key("abc"), new Key("2"), value2);

        Key duplicateOfKey1 = new Key("1");
        Key duplicateOfValue2 = new Key("xyz");

        biMap.forcePut(duplicateOfKey1, duplicateOfValue2);

        Assert.assertSame(key1, Iterate.getFirst(biMap.entrySet()).getKey());
        Assert.assertSame(key1, Iterate.getFirst(biMap.inverse().entrySet()).getValue());

        Assert.assertSame(value2, Iterate.getFirst(biMap.entrySet()).getValue());
        Assert.assertSame(value2, Iterate.getFirst(biMap.inverse().entrySet()).getKey());
    }`